### DISCUSSION TITLE
Is there a way to set network proxy for vcpkg？

### DISCUSSION THREAD

**Discussion by solayw** [microsoft/vcpkg#23082] [Q&A]

**Comment by ras0219-msft** [comment]
Hi @Adam74211,

Assuming you're talking about during the builds of libraries themselves, we currently use CMake to perform the downloads (which uses Curl underneath). Curl can be configured with the `HTTP_PROXY` and `HTTPS_PROXY` environment variables such as given by [1].

[1] https://stackoverflow.com/questions/9445489/performing-http-requests-with-curl-using-proxy

**Comment by paedocypris** [comment]
I had an issue in my company regarding proxy and wondered if the solution could be helpful to others.

Setting the environment variables didn't do the trick. After searching and realizing that vcpkg uses curl, I then tried to issue a curl command directly with "curl -q -I https://github...", which presented the error "curl: (4) unsupported proxy 'https://xxx:yyy@proxy.com:1234', libcurl is built without the HTTPS-proxy support".

Then, I just updated the HTTP_PROXY and HTTPS_PROXY variables to http://xxx:yyy@proxy.com:1234' and it worked as intended.
I hope it helps somebody else.

**Comment by LeoYawoo** [comment]
recently, i got this issue when use clion with vcpkg plugin, i make a stupid proxy way by use a wrapper exec process.
```
package main

import (
	"fmt"
	"log"
	"os"
	"os/exec"
	"path/filepath"
)

func main() {
	dir, err := filepath.Abs(filepath.Dir(os.Args[0]))
	if err != nil {
		log.Fatal(err)
	}
	// fmt.Println(dir)
	cmd := exec.Command("./vcpkg_raw.exe", os.Args[1:]...)
	cmd.Env = os.Environ()
	cmd.Env = append(cmd.Env, "http_proxy=http://127.0.0.1:xxx")
	cmd.Env = append(cmd.Env, "https_proxy=http://127.0.0.1:xx")
	cmd.Dir = dir
	cmd.Stdin = os.Stdin
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	err = cmd.Run()
	if err != nil {
		log.Fatal(err)
	}
}
```
a real ugly way lol.
